update/application.go

```diff
@@ -117,4 +126,23 @@ func (cmd *applicationCmd) applyUpdates(app *apps.Application) {
 	if cmd.BasicAuth != nil {
 		app.Spec.ForProvider.Config.EnableBasicAuth = cmd.BasicAuth
 	}
+	if cmd.DeployJob != nil {
+		existingDeployJob := app.Spec.ForProvider.Config.DeployJob
+		if existingDeployJob == nil {
+			app.Spec.ForProvider.Config.DeployJob = &apps.DeployJob{}
+		}
```

Comment:
This block of code can produce some unexpected results. 

Case 1
As I described it above, `cmd.DeployJob` is never nil. So let's say we have an `Application` created with no `spec.forProvider.config.deployJob` specified, for example:
```yaml
spec:
  forProvider:
    config:
      env: []
      port: null
      replicas: null
      size: ""
```

now, I am updating something which is _not_ part of the `deployJob`, for example `config.env`:
```bash
> nctl update app "destined-supergirl" --env=mykey1=myvalue1
```

this will update env variables (as expected) but also will create the following `spec.forProvider.config.deployJob`:
```yaml
spec:
  forProvider:
    config:
      deployJob:
        command: ""
        name: ""
        retries: 0
        timeout: 5m
      env:
      - name: mykey1
        value: myvalue1
```

Case 2
Let's say we have an `Application` created with no `spec.forProvider.config.deployJob` specified, for example:
```yaml
spec:
  forProvider:
    config:
      env: []
      port: null
      replicas: null
      size: ""
```

Now, I am updating `deployJob`, like so:
```bash
> nctl update app "cuddly-infragirl"  --deploy-job-command="nop"
```

this will create the following `spec.forProvider.config.deployJob`:
```yaml
spec:
  forProvider:
    config:
      deployJob:
        command: nop
        name: ""
        retries: 0
        timeout: 5m
      env: []
      port: null
      replicas: null
      size: ""
```
so again, an empty `name`
(assuming we will remove the default for `Name`:
```golang
type deployJob struct {
	Name    string        `default:"release" ....
}
```
)

Reply:
I have changed this to only create the `DeployJob` struct if it's nil *and* one of the flags is set.